identity/app/model/ReturnJourney.scala

```diff
@@ -0,0 +1,58 @@
+package model
+
+import com.netaporter.uri.Uri
+
+import scala.util.Try
+
+object ReturnJourney {
+  private val paymentFailureCodes = Set("PF", "PF1", "PF2", "PF3", "PF4", "CCX")
+  private val subscriptionsClientId = "subscriptions"
+  private val contributionsClientId = "recurringContributions"
+
+  sealed trait Journey {
+    def applies(returnUri: Uri): Boolean
+    val continueCopy: String = "Continue"
+  }
+
+  // Checks for INTCMP parameter in returnUrl after manage.theguardian redirects to signin with this parameter for payment failure flows
+  // If parameter exists and is for payment failure, the 'continue' button reads 'Update your payment method'
+  case object PaymentFailure extends Journey {
+    override def applies(returnUri: Uri): Boolean =
+      returnUri.query
+        .param("INTCMP")
+        .exists(paymentFailureCodes.contains)
+
+    override val continueCopy: String = "Update your payment method"
+  }
```

Comment:
This kind of logic really belongs in identity-frontend, where the clientId parameters mean something.  Until it's migrated I thought it better to pull it out into the identity "model" in frontend.